File: spring-boot-java-web/src/main/java/com/example/webjsp/service/impl/CityServiceImpl.java
package com.example.webjsp.service.impl;

import com.example.webjsp.mapper.CityMapper;
import com.example.webjsp.model.City;
import com.example.webjsp.service.CityService;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.HashMap;
import java.util.List;
import java.util.UUID;

@Service
@Slf4j
@Transactional
public class CityServiceImpl implements CityService {

    private final CityMapper cityMapper;

    public CityServiceImpl(@Autowired CityMapper cityMapper) {
        this.cityMapper = cityMapper;
    }

    @Override
    public City selectCity(UUID id) {
        City city = cityMapper.selectCity(id);
        log.debug("select-city: {}", city);
        return city;
    }

    @Override
    public List<HashMap<String, Object>> selectAll() {
        List<HashMap<String, Object>> rslt = cityMapper.selectAll();
        log.debug("select-all dump: {}", rslt);

        return rslt;
    }

    @Override
    public int selectCount() {
        int cnt = cityMapper.selectCount();
        log.debug("select-count: {}",cnt);
        return cnt;
    }
}

File: spring-boot-kotlin-web/src/main/kotlin/com/example/webjsp/service/impl/CityServiceImpl.kt
package com.example.webjsp.service.impl

import com.example.webjsp.mapper.CityMapper
import com.example.webjsp.model.City
import com.example.webjsp.service.CityService
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import java.util.*

@Service
@Transactional
class CityServiceImpl: CityService {

    @Autowired
    lateinit var cityMapper: CityMapper

    override fun selectCity(id: UUID): City {
        val v = cityMapper.selectCity(id)
        log.debug("select-city: {}", v)

        return v
    }

    override fun selectAll(): List<HashMap<String, Any>> {
        val src = cityMapper.selectAll()
        log.debug("select-all dump: {}", src)
        return src
    }

    override fun selectCount(): Int {
        val cnt = cityMapper.selectCount()
        log.debug("select-count: {}", cnt)
        return cnt
    }

    companion object {
        val log: Logger = LoggerFactory.getLogger(CityServiceImpl::class.java)
    }
}